plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -500,7 +508,14 @@ func (r *ReferenceManager) Admit(ctx context.Context, a admission.Attributes, _
 				wg.Add(len(shootList))
 
 				for _, s := range shootList {
-					if s.Spec.CloudProfileName != cloudProfile.Name || s.DeletionTimestamp != nil {
+					var cloudProfileName string
+					if s.Spec.CloudProfile != nil {
```

Comment:
```suggestion
					if s.Spec.CloudProfile != nil && s.Spec.CloudProfile.Kind == v1beta1constants.CloudProfileReferenceKindCloudProfile {
```

Not necessarily required now, but probably good if you are already on it.